hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/SparkFileFormatInternalRowReaderContext.scala

```diff
@@ -62,7 +63,7 @@ class SparkFileFormatInternalRowReaderContext(baseFileReader: Option[Partitioned
                                      requiredSchema: Schema,
                                      conf: Configuration): ClosableIterator[InternalRow] = {
     val fileInfo = sparkAdapter.getSparkPartitionedFileUtils
-      .createPartitionedFile(partitionValues, filePath, start, length)
+      .createPartitionedFile(InternalRow.empty, filePath, start, length)
```

Comment:
Ok so part of this PR is to clean up the philosophy of the fg reader.
We want to give the fg reader a requested schema, and we want the output to be an iterator with records in exactly that schema (maybe not with cdc, haven't given it much thought yet).
spark parquet file reader will append the partition values to the end of each record.  Putting the logic for dealing with that inside of the fg reader is adding unnecessary stuff into the fg reader that is probably only relevant for spark.
Therefore I think it makes sense for HoodieFileGroupReaderBasedParquetFileFormat to be responsible for appending the partition columns to the end. In addition to being better organized in my opinion, this will also be more performant because it prevents adding several calls that append the partition col and then project the column away.

Reply:
Good, I thought about that before. Meanwhile, the partitionValues should not be passed from context also

Reply:
@jonvex nit: add a comment/doc in the code to explain why the partition value is empty.